packages/plugin-client-drizzle/test/integration.ava.ts

```diff
@@ -0,0 +1,2187 @@
+test('select from raw sql', async (t) => {
+  const { db } = t.context;
+
+  const result = await db
+    .select({
+      id: sql<number>`id`,
+      name: sql<string>`name`
+    })
+    .from(sql`(select 1 as id, 'John' as name) as users`);
+
+  Expect<Equal<{ id: number; name: string }[], typeof result>>;
+
+  t.deepEqual(result, [{ id: 1, name: 'John' }]);
+});
+
+test.skip('select from raw sql with joins', async (t) => {
+  const { db } = t.context;
+
+  const result = await db
+    .select({
+      id: sql<number>`users.id`,
+      name: sql<string>`users.name`,
+      userCity: sql<string>`users.city`,
+      cityName: sql<string>`cities.name`
+    })
+    .from(sql`(select 1 as id, 'John' as name, 'New York' as city) as users`)
+    .leftJoin(sql`(select 1 as id, 'Paris' as name) as cities`, sql`cities.id = users.id`);
+
+  Expect<Equal<{ id: number; name: string; userCity: string; cityName: string }[], typeof result>>;
+
+  t.deepEqual(result, [{ id: 1, name: 'John', userCity: 'New York', cityName: 'Paris' }]);
+});
+
+test.skip('join on aliased sql from select', async (t) => {
+  const { db } = t.context;
+
+  const result = await db
+    .select({
+      userId: sql<number>`users.id`.as('userId'),
+      name: sql<string>`users.name`,
+      userCity: sql<string>`users.city`,
+      cityId: sql<number>`cities.id`.as('cityId'),
+      cityName: sql<string>`cities.name`
+    })
+    .from(sql`(select 1 as id, 'John' as name, 'New York' as city) as users`)
+    .leftJoin(sql`(select 1 as id, 'Paris' as name) as cities`, (cols) => eq(cols.cityId, cols.userId));
+
+  Expect<Equal<{ userId: number; name: string; userCity: string; cityId: number; cityName: string }[], typeof result>>;
+
+  t.deepEqual(result, [{ userId: 1, name: 'John', userCity: 'New York', cityId: 1, cityName: 'Paris' }]);
+});
+
+test.skip('join on aliased sql from with clause', async (t) => {
+  const { db } = t.context;
+
+  const users = db.$with('users').as(
+    db
+      .select({
+        id: sql<number>`id`.as('userId'),
+        name: sql<string>`name`.as('userName'),
+        city: sql<string>`city`.as('city')
+      })
+      .from(sql`(select 1 as id, 'John' as name, 'New York' as city) as users`)
+  );
+
+  const cities = db.$with('cities').as(
+    db
+      .select({
+        id: sql<number>`id`.as('cityId'),
+        name: sql<string>`name`.as('cityName')
+      })
+      .from(sql`(select 1 as id, 'Paris' as name) as cities`)
+  );
+
+  const result = await db
+    .with(users, cities)
+    .select({
+      userId: users.id,
+      name: users.name,
+      userCity: users.city,
+      cityId: cities.id,
+      cityName: cities.name
+    })
+    .from(users)
+    .leftJoin(cities, (cols) => eq(cols.cityId, cols.userId));
+
+  Expect<Equal<{ userId: number; name: string; userCity: string; cityId: number; cityName: string }[], typeof result>>;
+
+  t.deepEqual(result, [{ userId: 1, name: 'John', userCity: 'New York', cityId: 1, cityName: 'Paris' }]);
```

Comment:
Both incorrect ID and the known issue we have: https://github.com/xataio/xata/issues/2572